frontend/src/components/header/Header.jsx

```diff
@@ -1,44 +1,90 @@
-import React from 'react';
+import React, { useState, useEffect, useRef } from 'react';
 import { ReactComponent as Logo } from 'assets/images/logo.svg';
+import { ReactComponent as Menu } from 'assets/images/menu.svg';
+import { ReactComponent as Close } from 'assets/images/close.svg';
 import { NavLink } from 'react-router-dom';
 import TelegramLogin from '../Telegram/TelegramLogin';
 import './header.css';
 
 function Header({ walletId, onConnectWallet, onLogout, tgUser, setTgUser }) {
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
+  const menuRef = useRef(null);
+  const buttonRef = useRef(null);
+
+  useEffect(() => {
+    function handleClickOutside(event) {
+      if (
+        menuRef.current &&
+        !menuRef.current.contains(event.target) &&
+        buttonRef.current &&
+        !buttonRef.current.contains(event.target)
+      ) {
+        setIsMenuOpen(false);
+      }
+    }
+
+    document.addEventListener('mousedown', handleClickOutside);
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
+  }, []);
```

Comment:
Please create separate custom hook `useClickOutside` and use it instead

Reply:
Alright